hw/ip/otbn/dv/verilator/otbn_top_sim.cc

```diff
@@ -180,6 +180,55 @@ extern "C" int OtbnTopInstallLoopWarps() {
   return 0;
 }
 
+template <typename PrimCountT>
+void set_up_down_prim_count(PrimCountT *prim_count, uint32_t new_cnt) {
+  // There are both up and down counters. The down counter gives the remaining
+  // iterations and the up counter is the inverse. They must always sum to a
+  // constant value given by `max_val`.
+
+  // Loop warping occurs on the negative clock edge. When writing signal values
+  // here that won't trigger any processes within the verilator simulation. At
+  // the next positive clock edge @(posedge clk) processes will run before any
+  // _d update processes so write the relevant _d values here.
+
+  // down_cnt doesn't have seperate _d and _q values. Instead the logic to
+  // update down_cnt is within the @always(posedge clk) that writes the flop.
+  prim_count->gen_cross_cnt_hardening__DOT__down_cnt = new_cnt;
+
+  auto up_cnt_flop = prim_count->gen_cnts__BRA__0__KET____DOT__u_cnt_flop;
```

Comment:
These are internal names. Is there a reason they can't be exposed as public symbols? I don't think the symbols you've written here are guaranteed stable across Verilator versions.

Reply:
I believe these are are the public names, see the waiver file where the relevant signals are declared public, which in turn makes the modules containing them public.

https://github.com/lowRISC/opentitan/blob/5b37a57a1986bc10437593175def370b99c7d502/hw/ip/otbn/dv/verilator/otbn_top_sim_waivers.vlt#L16-L23

The actual hierarchy path is gen_cnts[0].u_cnt_flop. Obviously not a valid C++ identifier so verilator uses __BRA__, __KET__ and __DOT__

Reply:
Ohh! Sorry, I didn't understand. Cool :-)